### ISSUE TITLE
closable tab crash

### ISSUE THREAD

**Issue by MoAlyousef** [fltk/fltk#775]
**Describe the bug**
Trying out the closable tab feature, it seems that recently it started crashing.

**To Reproduce**
This is a minimal reproducible example:
```cpp
#include <FL/Fl_Tabs.H>
#include <FL/Fl.H>
#include <FL/Fl_Group.H>
#include <FL/Enumerations.H>
#include <FL/Fl_Double_Window.H>

void tab_close_cb(Fl_Widget *w, void *) {
    if (Fl::callback_reason() == FL_REASON_CLOSED) {
        auto parent = w->parent();
        parent->remove(w);
    }
}

int main() {
    auto win = new Fl_Double_Window(400, 300);
    auto tabs = new Fl_Tabs(0, 0, 400, 300);
    auto grp1 =  new Fl_Group(0, 30, 400, 300 - 30, "Tab1");
    grp1->when(FL_WHEN_CLOSED);
    grp1->callback(tab_close_cb);
    grp1->end();
    auto grp2 = new Fl_Group(0, 30, 400, 300 - 30, "Tab2");
    grp2->when(FL_WHEN_CLOSED);
    grp2->callback(tab_close_cb);
    grp2->end();
    tabs->end();
    win->end();
    win->show();
    return Fl::run();
}
```

Build the example with latest master version. For the example I built FLTK from source and compiled the example using `fltk-config --compile main.cpp`
After the window shows, close the tabs and you will get a segfault.

**Expected behavior**
To not crash

**FLTK Version**
 - Version: 1.4.0
 - Downloaded and self-built from: git
 - If from Git, branch: master

**FLTK Configure / Build Options**
My locally built FLTK is built with cmake with uses -DFLTK_BUILD_TEST=OFF.

**Operating System / Platform:**
 - OS: Kubuntu 20.04 x86_64, running on a machine. Not virtualized.

**Linux/Unix Runtime, if applicable:**
 - X11

**Additional context**
I discovered this when the corresponding Rust example also crashed, so I `git pull`ed my local copy of FLTK from master, and it seems something has changed. As previously the equivalent Rust code didn't crash on closing all tabs.
In the C++ repro, I tried replacing the parent->remove with deleting the widget using Fl::delete_widget or delete (Fl_Group*)w, but it appears that has not changed the behavior.

**Comment by Albrecht-S** [comment]
Thanks for the report. It crashes in `draw()` when all tabs are removed. It would very likely also crash if you created an `Fl_Tabs` widget w/o children in the first place (although I didn't test this case explicitly).

The crash is in [src/Fl_Tabs.cxx, line 796](https://github.com/fltk/fltk/blob/a683752e4b312530067e9973f489edf9476db744/src/Fl_Tabs.cxx#L796) because `tab_pos` is NULL when no children are present. A "brute force" fix is to wrap this entire block in
```
  if (children()) {
    ...
  }
```

This worked for me but I'm leaving the final fix to @MatthiasWM who wrote this part of the code. TIA.

**Comment by Albrecht-S** [comment]
@MatthiasWM I assigned this issue to you (see my comment above) and set labels "`bug`" and "`Prio 3 - Moderate`" because this case is an unusual (although legit) usage of `Fl_Tabs`.

**Comment by MatthiasWM** [comment]
Thanks, I will try to get to this today.

**Comment by MatthiasWM** [comment]
Should be fixed now: [5ef2920](https://github.com/fltk/fltk/commit/5ef29202ede1042da22044bb10017e2db04a89df).

Please close the issue after verification.

**Comment by MoAlyousef** [comment]
Thank you. I confirm the issue is fixed.
